engine/gamesys/src/gamesys/components/comp_collection_factory.cpp

```diff
@@ -35,21 +36,38 @@ DM_PROPERTY_U32(rmtp_CollectionFactory, 0, FrameReset, "# components", &rmtp_Com
 
 namespace dmGameSystem
 {
-    using namespace dmVMath;
+    struct CollectionFactoryResource;
 
     const char* COLLECTION_FACTORY_MAX_COUNT_KEY = "collectionfactory.max_count";
 
+    static const dmhash_t COLLECTION_FACTORY_PROP_PROTOTYPE = dmHashString64("prototype");
+
     static void CleanupAsyncLoading(lua_State*, CollectionFactoryComponent*);
     static bool PreloadCompleteCallback(const dmResource::PreloaderCompleteCallbackParams*);
     static void LoadComplete(const dmGameObject::ComponentsUpdateParams&, CollectionFactoryComponent*, const dmResource::Result);
     static dmResource::Result LoadCollectionResources(dmResource::HFactory, CollectionFactoryComponent*);
     static void UnloadCollectionResources(dmResource::HFactory, CollectionFactoryComponent*);
+    static CollectionFactoryResource* GetResource(CollectionFactoryComponent* component);
+
+    struct CollectionFactoryComponent
```

Comment:
In general, we want our structs private so that we are forced to keep our scripting clean, and code complexity down to a minimum.